### PR TITLE
Rework GitHub acceptor and move to its own module

### PR DESCRIPTION
**Pull request by paulmelnikow**
Continuing to merge the work from #1205.

### CODE REVIEW

**Comment by paulmelnikow** [comment]
This really should have a test. I'm eager to get it in.

File: services/github/auth/acceptor.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          })
+        })
+    )
+  )
+}
+
+function setRoutes(server) {
+  const baseUrl = process.env.BASE_URL || 'https://img.shields.io'
+
+  server.route(/^\/github-auth$/, (data, match, end, ask) => {
```
Comment (paulmelnikow) [inline services/github/auth/acceptor.js]: Added a test.

File: services/github/auth/acceptor.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      redirect_uri: `${baseUrl}/github-auth/done`,
+    })
+    ask.res.setHeader(
+      'Location',
+      `https://github.com/login/oauth/authorize?${query}`
+    )
+    end('')
+  })
+
+  server.route(/^\/github-auth\/done$/, (data, match, end, ask) => {
```
Comment (paulmelnikow) [inline services/github/auth/acceptor.js]: Added a test.

**Comment by paulmelnikow** [comment]
Thanks for reviewing! Yea, not sure why package-lock didn't get updated. I will do that now.

**Comment by paulmelnikow** [comment]
Yea, it's a consistent problem with dependabot. The churn occurs every time a package-lock is written by the npm CLI after it's been written by dependabot, or vice versa. I'm not sure we've reported it. It sure would be nice to fix because it makes it basically impossible to review the changes in that file.

**Comment by chris48s** [comment]
The lockfile optional thing is a reported issue on dependabot https://github.com/dependabot/feedback/issues/197 but required an upstream fix: https://github.com/npm/cli/pull/76

As I understand it, once we (and dependabot) are updated to a version on npm with that patch, this should stop happening.

**Pull request merged**